packages/ssh-tunnel/src/index.ts

```diff
@@ -195,7 +145,55 @@ export class SshTunnel extends EventEmitter {
     }
   }
 
+  private async connectSsh(stayConnected = false): Promise<void> {
+    this.stayConnected = stayConnected || this.stayConnected;
+
+    if (this.connected) {
+      debug('already connected');
+      return;
+    }
+
+    if (this.connectingPromise) {
+      debug('reusing connectingPromise');
+      return this.connectingPromise;
+    }
+
+    if (!stayConnected) {
```

Comment:
super NIT, I would have probably this variable easier if this variable was called something like `closed`:

```
if (closed) {
   throw new Error('Tunnel already closed.');
}
```